### DISCUSSION TITLE
FitAddon is calculating wrong dimension

### DISCUSSION THREAD

**Discussion by bajrangCoder** [sourcelair/xterm.js#4853] [Q&A] [answered]
I am using fitaddon to fit terminal in my element and my elements height , width is set to 100% but fit addon calculates small dimensions of terminal as shown in given image.
And I am using xtermjs: v5.2.1 and fitaddon: v0.7.0

![Screenshot_20231023-152853~3](https://github.com/xtermjs/xterm.js/assets/71929976/49a4e21c-e3b9-4d88-8ec4-a4f5150a2080)

**Comment by tisilent** [comment]
~~There's a more violent way.~~
~~`const resizeObserver = new ResizeObserver(entries => {
      addons.fit.instance.fit();
  });
  resizeObserver.observe(terminalContainer);`~~

~~terminalContainer -> open dom~~

Wrong understanding.

**Comment by tisilent** [comment]
@bajrangCoder hi. Can you provide CSS for your elements.

**Comment by jerch** [comment, marked as answer]
@bajrangCoder Bottom gap looks likes its smaller than one line in height, right side gap looks like its reserving space for the scrollbar + a gap < 1 cell width - I'd say this works as intended.

Those cell fitting gaps can only be leveled out by some CSS adjusting rules, the terminal/fit addon cannot fill that fully, if the dimensions of the outer container is not a perfect fit.

Regarding the scrollbar space - this can be deactivated (makes sense if your content never gonna run it scrollback), or just always show it.

**Reply by bajrangCoder** [reply]
Ok got it
